### ISSUE TITLE
[KEYBOARD]: Community Care Service Type does not warn users when field is reset because search did not match a term

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#489]
## Issue
The Community Care service type typeahead requires users to select a type like "primary care" or "cardiology" from the dropdown menu. If users enter a search term like "test" and tab away from the field, it is reset to the empty default. We do not throw an error or warning to users, and this was noted as an SC 3.3.1 violation. Animated GIF attached below.

This may require some design input, because we're asking for an error message in conjunction with a longer label (see related issue). CC'ing @emilywaggoner for awareness.

## Related Issue
https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/488

## Audit Finding
* Note 6, Defect 1 of 1

## Acceptance criteria
[ ] Service Type field shows a clearly marked error message when the field resets to default or user has entered data incorrectly.
[ ] Error is read aloud when the Service Type field is reset to default or an assistive device user has entered data incorrectly.
[ ] Run automated scans periodically during development
[ ] Run the axe-core plugin while developing new components. Many automated errors can be avoided or fixed during the development cycle.
[ ] Internal peer review 
[ ] After peer review, share with the FL team for Product/Design sign off and/or FE Review Team 
[ ]  Documentation and points of contact for any new backend dependencies
[ ]  Links to important dashboards for investigating relevant issues

## Definition of done
[ ] Events are logged appropriately
[ ] Documentation has been updated, if applicable
[ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
[ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Environment
* MacOS Mojave
* Chrome latest
* https://staging.va.gov/find-locations/

## WCAG or Vendor Guidance (optional)
* [Error Identification: Understanding SC 3.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html)

## Screenshots or Trace Logs
![no-warning-on-empty.gif](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/537bfc81-0f5b-4b42-930e-c7626a88e837)

**Comment by mmiddaugh** [comment]
Need to confirm this has not been resolved from 508 compliance side
@jenstrickland , can you retest to see if the error is picked up by assistive devices?

**Comment by jenstrickland** [comment]
@mmiddaugh This issue still exists. To test:

- Go to [https://staging.va.gov/find-locations/](https://staging.va.gov/find-locations/)

- Type in a city, state, or Zip code, or leave the `Search city, state, or postal code` blank

- In the `Search for` dropdown, select Community Care

- Tab to the next field and type `test` or any other combination of characters then tab out of the field.

- The field defaults to the original text with no error message. 

We need to provide an error or warning to users. 

No assistive tech is required to test this issue, initially, however the solution will need to pass the Acceptance Criteria, providing both a visual error and one that may be read by screen readers.

**Comment by mmiddaugh** [comment]
We may want to consider [487](https://github.com/department-of-veterans-affairs/va.gov-team/issues/489) when working through this issue.

**Comment by davidconlon** [comment]
@jenstrickland just to clarify, would it be appropriate to rephrase as "Rather than _only_ disable the Search button, we want to provide error messages when required fields are not complete"

@mmiddaugh there is a possibility that [487](https://github.com/department-of-veterans-affairs/va.gov-team/issues/487) is being triggered by a completely different script. Might make sense to pair these, but 487 may require a little more research to determine root cause.

**Comment by jenstrickland** [comment]
@davidconlon 
The latest I’ve been hearing is not to disable buttons, and to handle errors inline whenever possible.

Our big forms that span multiple pages don’t disable the Continue button, they just fire error messages on the submit event if users have skipped past required fields (forms like healthcare, 526 (disability increase), education benefits).

Since we don't mark our required fields on this form, it may be difficult for people to realize the inputs are required. If the submit button is disabled, a person may look at the form and not know why, just keep trying to click it. If the button is enabled, then the person gets an error message that lets them know what needs attention.

**Comment by davidconlon** [comment]
Apologies. I didn't see the thread in [3050](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/3050) that explained all of this. Michelle closed that other issue and updated the acceptance criteria.  Thanks!

**Comment by mmiddaugh** [comment]
This acceptance criteria has been added to [#4322] and this issue will be closed.
